src/coreclr/nativeaot/System.Private.CoreLib/src/System.Private.CoreLib.csproj

```diff
@@ -331,12 +329,25 @@
       <Link>Interop\Unix\System.Native\Interop.MUnmap.cs</Link>
     </Compile>
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsBrowser)'=='true' and '$(FeatureWasmThreads)' != 'true'">
-    <Compile Include="System\Threading\PreAllocatedOverlapped.NativeAot.Wasm.cs" />
+
+  <!-- WASM-specific threading things. Keep in sync with Mono. -->
+  <PropertyGroup>
+    <MonoBclSourcesRoot>$([MSBuild]::NormalizePath('$(MonoProjectRoot)', 'System.Private.CoreLib', 'src'))</MonoBclSourcesRoot>
+  </PropertyGroup>
+  <ItemGroup Condition="'$(TargetsBrowser)' == 'true' and '$(FeatureWasmThreads)' != 'true'">
+    <Compile Include="$(MonoBclSourcesRoot)\System\Threading\ThreadPoolBoundHandle.Browser.Mono.cs" />
+    <Compile Include="$(MonoBclSourcesRoot)\System\Threading\PreAllocatedOverlapped.Browser.Mono.cs" />
+    <Compile Include="$(MonoBclSourcesRoot)\System\Threading\ThreadPool.Browser.Mono.cs" />
+    <!-- TODO-LLVM: the timer queue is a little more dependent on TS, take the PNSE version for now. -->
+    <Compile Include="$(MonoBclSourcesRoot)\System\Threading\TimerQueue.Wasi.Mono.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsWasi)'=='true'">
-    <Compile Include="System\Threading\ThreadApiCompat.Wasi.cs" />
+  <ItemGroup Condition="'$(TargetsWasi)' == 'true' and '$(FeatureWasmThreads)' != 'true'">
+    <Compile Include="$(MonoBclSourcesRoot)\System\Threading\ThreadPoolBoundHandle.Browser.Mono.cs" />
```

Comment:
This reads a little strange: including "Browser" files in Wasi.  I suppose its unavoidable right now.

Reply:
The whole implementation is just a PNSE stub; I suppose it just wasn't renamed to something like `.WebAssembly.` at the time WASI was being brought up.